autopush/web/registration.py

```diff
@@ -0,0 +1,311 @@
+import json
+import re
+import time
+import uuid
+
+from autopush.exceptions import InvalidRequest
+from autopush.web.base import threaded_validate, BaseWebHandler, PREF_SCHEME, \
+    AUTH_SCHEMES
+from autopush.db import (
+    generate_last_connect,
+    hasher,
+)
+from autopush.utils import generate_hash, ms_time
+from boto.dynamodb2.exceptions import ItemNotFound
+from cryptography.hazmat.primitives import constant_time
+from marshmallow import Schema, fields, pre_load, validates_schema
+from twisted.internet.defer import Deferred
+from twisted.internet.threads import deferToThread
+
+
+class RegistrationSchema(Schema):
+    uaid = fields.Str(allow_none=True)
+    chid = fields.Str(allow_none=True)
```

Comment:
We can switch UAID over to UUID, but I think we'll need to keep CHID as a Str since that could be sent by the client.